### PR TITLE
Texture support for raw `Mesh3D` logging

### PR DESCRIPTION
**Pull request by Wumpf**
### What

Meshes can now be logged with optional texture coordinates and an optional albedo texture!

Vertex texture coordinates are straight forward, implemented as a new component, mostly to avoid semantic confusion (position2d on mesh?) and being able to call the coordinates `uv`.

~The (albedo) texture itself got a bit messy: The original plan was to refer to a separately logged image and I believe that's still where we want to go. But this messes with some parts of the viewer so I made the decision to put a `TensorData` based field on the material. The main drawback of that is that now you can't inspect the texture separately and the format restrictions are quite tight (this is in part also due to the mesh renderer itself though, so that's deeper than just not using the Image archetype). I believe in the future the material should instead have an entity reference to specify where the texture comes from.
Another pretty bad sideffect from this is that if you have model with a bunch of meshes using the same texture several times, it will send this texture each time to the viewer since there's no way to "instance" this.~

~... but let's perfect not be the enemy of well-it-works-for-the-most-important-usecases-so-let's-ship-this!~

Change: Texture is now a component, making a lot of things nicer

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/4894/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/4894/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/4894/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4894)
- [Docs preview](https://rerun.io/preview/cf2f654b4ebec5348300032f7ec7c0494d3663e4/docs) 
- [Examples preview](https://rerun.io/preview/cf2f654b4ebec5348300032f7ec7c0494d3663e4/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Comment by teh-cmc** [comment]
I cannot compile the rust example:
```sh
$ cargo r -q -p raw_mesh
error[E0432]: unresolved imports `rerun::external::ecolor`, `rerun::external::re_memory`
  --> examples/rust/raw_mesh/src/main.rs:16:16
   |
16 |     external::{ecolor, re_log, re_memory::AccountingAllocator},
   |                ^^^^^^          ^^^^^^^^^ could not find `re_memory` in `external`
   |                |
   |                no `ecolor` in `external`

error[E0433]: failed to resolve: could not find `clap` in `rerun`
   --> examples/rust/raw_mesh/src/main.rs:126:19
    |
126 |     rerun: rerun::clap::RerunArgs,
    |                   ^^^^ could not find `clap` in `rerun`

error[E0425]: cannot find function `setup_native_logging` in crate `re_log`
   --> examples/rust/raw_mesh/src/main.rs:182:13
    |
182 |     re_log::setup_native_logging();
    |             ^^^^^^^^^^^^^^^^^^^^ not found in `re_log`

Some errors have detailed explanations: E0425, E0432, E0433.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `raw_mesh` (bin "raw_mesh") due to 3 previous errors
```

**Comment by teh-cmc** [comment]
Python example works great but there are some issues in the data views.

In the outer data view, the material shows up as `(empty)`:
image

In the data view of the material itself:
- The `albedo_factor` shows up as `(empty)` which is... weird? :thinking: what does it mean for an albedo factor to be empty?
- The texture shows its dimensions but there's no way to preview it. Any chance we can re-use the existing preview machinery for images/tensors?
image

File: examples/rust/raw_mesh/src/main.rs
Comment (teh-cmc) [inline examples/rust/raw_mesh/src/main.rs:58]: Why not do it for the rust example too?

**Comment by Wumpf** [comment]
> Would be nice to update the screenshots in raw_mesh readmes (python & rust).

Unfortunately I didn't update the Rust example (and apparently also broke it :/) though to upload the texture. Guess I shouldn't leave this business unfinished

File: examples/rust/raw_mesh/src/main.rs
Comment (Wumpf) [reply examples/rust/raw_mesh/src/main.rs:58]: I ran out of time and didn't see a quick & easy path towards that. The gltf loader doesn't load the image itself, so I need an extra path that loads the images from disk etc.

**Comment by Wumpf** [comment]
> The albedo_factor shows up as (empty) which is... weird? 🤔 what does it mean for an albedo factor to be empty?

Albedo factor showed like that before when it's null. Don't know what else to write there. Isn't `(empty)` what we usually show for null

> The texture shows its dimensions but there's no way to preview it. Any chance we can re-use the existing preview machinery for images/tensors?

Yeah this sucks, but I found this really hard to fix that right now with the setup of the texture not being a `Image` archetype. See code comments & pr description. Again, this was outside my time budget

**Review by teh-cmc — APPROVED** [review]
Works great and I agree with everything you said :+1: :ship:

File: crates/re_types/definitions/rerun/datatypes/material.fbs
Comment (Wumpf) [reply crates/re_types/definitions/rerun/datatypes/material.fbs:23]: ah, what I meant to say is that it's sRGB(A)! In any case good you pointed out that this is confusing!

File: crates/re_space_view_spatial/src/visualizers/meshes.rs
Comment (Wumpf) [reply crates/re_space_view_spatial/src/visualizers/meshes.rs:115]: uhhh excellent find. Yeah let's do what you suggest with the salt

Comment (Wumpf) [reply crates/re_space_view_spatial/src/visualizers/meshes.rs:115]: actually using the mesh cache key directly also should do the trick

File: crates/re_space_view_spatial/src/mesh_loader.rs
Comment (emilk) [inline crates/re_space_view_spatial/src/mesh_loader.rs:236]: I suggest we re-export common types that are part of our public API

```suggestion
            format: re_renderer::TextureFormat::Rgba8UnormSrgb,
```

Comment (Wumpf) [reply crates/re_space_view_spatial/src/mesh_loader.rs:236]: in other places we tended to directly import wgpu instead 🤔. I have a bit of a bad feeling about exposing `TextureFormat` directly given that re_renderer leaks a lot of wgpu types in a lot of its APIs - I'm unsure where to stop.

Comment (emilk) [reply crates/re_space_view_spatial/src/mesh_loader.rs:236]: I would make the cutoff at "very commonly used" or not, but 🤷  your call

**Comment by Wumpf** [comment]
Blocked by a crash in arrow when logging only a color and no image

**Comment by Wumpf** [comment]
Blocked by
* #4971

File: crates/re_space_view_spatial/src/mesh_cache.rs
Comment (teh-cmc) [inline crates/re_space_view_spatial/src/mesh_cache.rs:30]: They don't live in the material anymore though.

```suggestion
        /// If there are any textures associated with that mesh (albedo etc), they use this hash for texture manager lookup.

**Comment by teh-cmc** [comment]
The fact that we cannot create a 2D view of the albedo texture when inspecting it because it is not in itself an entity is an interesting and extremely frustrating limitation of our current system.

I'll probably open an issue actually...

image

File: examples/rust/raw_mesh/src/main.rs
Comment (teh-cmc) [inline examples/rust/raw_mesh/src/main.rs:34]: that is in fact not done; we really should do it at some point

**Review by teh-cmc — APPROVED** [review]

**Pull request merged**
